frontend/src/components/Forms/RegisterPage/RegisterPage.tsx

```diff
@@ -0,0 +1,167 @@
+    });
+
+    if (!passwordMatch) {
+      setPasswordMatch(true);
+    }
+    if (!usernameAvaliable) {
+      setUsernameAvaliable(true);
+    }
+    if (usernameEmpty) setUsernameEmpty(false);
+    if (passwordEmpty) setPasswordEmpty(false);
+    if (confirmPasswordEmpty) setConfirmPasswordEmpty(false);
+    if (!passwordValid) setPasswordValid(true);
+
+    usernameField.current?.classList.remove(classes.inputError);
+    passwordField.current?.classList.remove(classes.inputError);
+    confirmPasswordField.current?.classList.remove(classes.inputError);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (isProcessing) return;
+
+    setIsProcessing(true);
+
+    if (formData.username === "") {
+      usernameField.current?.classList.add(classes.inputError);
+      return setUsernameEmpty(formData.username === "");
+    }
+    if (formData.password === "") {
+      passwordField.current?.classList.add(classes.inputError);
+      return setPasswordEmpty(formData.password === "");
+    }
+    if (formData.confirmPassword === "") {
+      confirmPasswordField.current?.classList.add(classes.inputError);
+      return setConfirmPasswordEmpty(formData.confirmPassword === "");
+    }
+    if (formData.confirmPassword !== formData.password) {
+      confirmPasswordField.current?.classList.add(classes.inputError);
+      return setPasswordMatch(false);
+    }
+
+    if (!passwordPattern.test(formData.password)) {
+      passwordField.current?.classList.add(classes.inputError);
+      confirmPasswordField.current?.classList.add(classes.inputError);
+      return setPasswordValid(false);
+    }
+    try {
+      const resp = await fetch("http://localhost:3000/register", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      if (resp.ok) {
+        console.log("success");
+        onClick();
+      } else {
+        console.log("failure");
+      }
+    } catch (err) {
+      console.log("Error: ", err);
+    } finally {
+      setIsProcessing(false);
+    }
+  };
+
+  return (
+    <div className={classes.container}>
+      <div className={classes.sheet}>
+        <button className={classes.close} onClick={onClick}>
+          <X />
+        </button>
+        <h1 className={classes.title}>Register!</h1>
+        <form className={classes.form} onSubmit={handleSubmit}>
+          <span className={classes.label}>Username</span>
+          <input
+            ref={usernameField}
+            className={classes.input}
+            name="username"
+            type="text"
+            value={formData.username}
+            onChange={handleChange}
+          />
+          {usernameEmpty && (
+            <div className={classes.warning} style={{ paddingBottom: "10px" }}>
+              Please enter your username
+            </div>
+          )}
+          {!usernameAvaliable && (
```

Comment:
'available', sorry to be pedantic (this typo seems to be present in a few places)